src/lib/hooks/useOnPress.js

```diff
@@ -1,14 +1,34 @@
-import { useCallback } from 'react'
+import { useCallback, useRef } from 'react'
```

Comment:
synced that hook with the RN version. have kept your change (checking for for .preventDefault)